saleor/graphql/webhook/utils.py

```diff
@@ -1,58 +1,69 @@
-from typing import Dict, Optional
+from typing import Dict, List, Optional
 
 from graphene.utils.str_converters import to_snake_case
-from graphql import parse, print_ast
-from graphql.error.syntax_error import GraphQLSyntaxError
+from graphql import parse
 from graphql.language.ast import Document
 
 
-def get_event_type_from_subscription(query: str) -> Optional[str]:
+def get_event_type_from_subscription(query: str) -> List[str]:
     try:
         ast = parse(query)
-        subscription = None
-        fragments = get_event_fragments_from_query(ast)
-
-        for definition in ast.definitions:
-            if (
-                hasattr(definition, "operation")
-                and definition.operation == "subscription"
-            ):
-                subscription = definition
-
-        if subscription:
-            subscription_string = print_ast(subscription)
-
-            if fragments:
-                for fragment_name, event_type in fragments.items():
-                    if fragment_name in subscription_string:
-                        return to_snake_case(event_type)
-
-            event_type = (
-                "".join(subscription_string.split())
-                .split("event{...on")[1]
-                .split("{")[0]
-            )
-            return to_snake_case(event_type)
-        return None
-    except (GraphQLSyntaxError, IndexError):
-        return None
-
-
-def get_event_fragments_from_query(ast: Document) -> Optional[Dict[str, str]]:
+        subscription = get_subscription(ast)
+        if not subscription:
+            return []
+
+        event_field = get_event_field_from_subscription(subscription)
+        if not event_field:
+            return []
+
+        events = get_events(event_field)
+        if not events:
+            return []
+
+        fragments = get_event_fragment_names_from_query(ast)
+        if fragments:
+            events = [fragments.get(event, event) for event in events]
+
+        return list(map(to_snake_case, events))
+
+    except Exception:
```

Comment:
Which part exactly can throw an exception? It's better to wrap small chunks with try..except and handle a specific exception instead of `Exception` which might hide some other errors.